### PR TITLE
FIX-#3182: Restricted upper numpy version because of potential bug in it

### LINKED ISSUE #3182: modin/pandas/test/test_io.py::TestCsv::test_read_csv_compression tests fail with numpy>1.20.3
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:

Ubuntu 20.04.1 LTS

- **Modin version** (`modin.__version__`):

0.10.0+6.gf9c99c38

- **Python version**:

Python 3.8.6

- **Code we can use to reproduce**:

Update numpy to the latest version (currently 1.21.0) and run modin/pandas/test/test_io.py::TestCsv::test_read_csv_compression tests. When they are ran together, tests with python engine fail with exception.

```
Traceback (most recent call last):
  File "/localdisk/gashiman/modin/modin/pandas/test/utils.py", line 668, in execute_callable
    pd_result = fn(pandas_df, **pd_kwargs)
  File "/localdisk/gashiman/modin/modin/pandas/test/utils.py", line 756, in applyier
    result = getattr(module, fn_name)(*args, **kwargs)
  File "/localdisk/gashiman/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers.py", line 610, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/localdisk/gashiman/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers.py", line 468, in _read
    return parser.read(nrows)
  File "/localdisk/gashiman/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers.py", line 1057, in read
    index, columns, col_dict = self._engine.read(nrows)
  File "/localdisk/gashiman/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers.py", line 2502, in read
    data = self._convert_data(data)
  File "/localdisk/gashiman/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers.py", line 2582, in _convert_data
    return self._convert_to_ndarrays(
  File "/localdisk/gashiman/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers.py", line 1704, in _convert_to_ndarrays
    cvals, na_count = self._infer_types(
  File "/localdisk/gashiman/miniconda3/envs/modin/lib/python3.8/site-packages/pandas/io/parsers.py", line 1765, in _infer_types
    na_count = isna(result).sum()
  File "/localdisk/gashiman/miniconda3/envs/modin/lib/python3.8/site-packages/numpy/core/_methods.py", line 48, in _sum
    return umr_sum(a, axis, dtype, out, keepdims, initial, where)
TypeError: int() argument must be a string, a bytes-like object or a number, not '_NoValueType'
```

### PR DESCRIPTION
**Pull request by gshimansky**
Signed-off-by: Gregory Shimansky <gregory.shimansky@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3182 
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: environment-dev.yml
Comment (vnlitvinov) [inline environment-dev.yml:6]:
```suggestion
# Restrict numpy upper version because of #3182
```

File: requirements-dev.txt
Comment (vnlitvinov) [inline requirements-dev.txt:2]:
```suggestion
# Restrict numpy upper version because of #3182
```

File: requirements/env_omnisci.yml
Comment (vnlitvinov) [inline requirements/env_omnisci.yml:9]:
```suggestion
# Restrict numpy upper version because of #3182
```

File: requirements/requirements-no-engine.yml
Comment (vnlitvinov) [inline requirements/requirements-no-engine.yml:5]:
```suggestion
# Restrict numpy upper version because of #3182
```

File: setup.py
Comment (vnlitvinov) [inline setup.py:63]:
```suggestion
# Restrict numpy upper version because of #3182
```

**Review by vnlitvinov — APPROVED** [review]
LGTM assuming tests pass.

The suggestions are only about making slightly better wording, we can live without them.

**Comment by gshimansky** [comment]
I think my wording is fine and perfectly convey the thought that I wanted to express. Waiting for another round of CI doesn't worth the hassle to change it, especially if we are going to remove this restriction in the future.

**Pull request merged**
